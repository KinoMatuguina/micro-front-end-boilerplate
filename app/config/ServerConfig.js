/**
* ServerConfig.js
*/

import { LOGGER } from '../utils/ServerLogger';
let ENV_CONFIG;

if (process.env.NODE_ENV === 'development') {
  ENV_CONFIG = require('../../webpack-envs/development');
}

if (process.env.NODE_ENV === 'production') {
  ENV_CONFIG = require('../../webpack-envs/production');
}

if (process.env.NODE_ENV === "qa") {
  ENV_CONFIG = require('../../webpack-envs/qa');

}

LOGGER.info("ENVIRONMENT CONFIG");
LOGGER.info(ENV_CONFIG);

const NODE_ENV = process.env.NODE_ENV || ENV_CONFIG.NODE_ENV;
const host = process.env.HOST || ENV_CONFIG.HOST || '0.0.0.0';
const apiPort = process.env.APIPORT || ENV_CONFIG.APIPORT;
const port = process.env.OVERR_PORT || process.env.PORT || ENV_CONFIG.PORT;
const apiHost = process.env.APIHOST || ENV_CONFIG.APIHOST || 'http://192.168.1.249';
const fileUploadHost = process.env.FILEUPLOAD_HOST || ENV_CONFIG.FILEUPLOAD_HOST || 'http://localhost:3030';
const context = process.env.APP_CONTEXT || ENV_CONFIG.APP_CONTEXT;
const securityLevel = process.env.SECURITY_LEVEL || ENV_CONFIG.SECURITY_LEVEL || 'HIGH';
const consulHost = process.env.CONSUL_HOST || ENV_CONFIG.CONSUL_HOST || '192.168.1.163';
const consulPort = process.env.CONSUL_PORT || ENV_CONFIG.CONSUL_PORT || '8500';
const authErrorRedirect = process.env.AUTH_ERR_REDIR || ENV_CONFIG.AUTH_ERR_REDIR || '/login/';
const redisHost = process.env.OVERR_REDIS_HOST || process.env.REDIS_HOST || ENV_CONFIG.REDIS_HOST || '192.168.1.249';
const redisPort = process.env.OVERR_REDIS_PORT || process.env.REDIST_PORT || ENV_CONFIG.REDIS_PORT || 6379;
const redisPassword = process.env.REDIS_PASSWORD || "12(5bo_)%"+ "\\" + "6!8BF";
const expressSessionSecret = process.env.EXPRESS_SESSION_SECRET || ENV_CONFIG.EXPRESS_SESSION_SECRET || '10qpalzm';
const expressSessionName = process.env.EXPRESS_SESSION_NAME || ENV_CONFIG.EXPRESS_SESSION_NAME || 'express.sid';
const timeout = process.env.TIMEOUT || ENV_CONFIG.TIMEOUT || 300
LOGGER.info(context);

module.exports = Object.assign({
  NODE_ENV: NODE_ENV,
  host: host,
  port: port,
  apiHost: apiHost,
  fileUploadHost: fileUploadHost,
  apiPort: apiPort,
  context: context,
  securityLevel: securityLevel,
  authErrorRedirect: authErrorRedirect,
  redis: {
    host: redisHost,
    port: redisPort,
    password: redisPassword
  },
  defaultTimeout: timeout,
  expressSession: {
    secret: expressSessionSecret,
    name: expressSessionName
  },
  defaultTimeout: timeout,
  consul: {
    enabled: true,
    host: consulHost,
    port: consulPort,
    promisify: true,
    keys: {
      globalMenu: 'config/application/global-menu',
      account: 'config/application/frontend-api-host/account',
      fundtransfer: 'config/application/frontend-api-host/fundtransfer',
      fileUploadHost: 'config/application/frontend-api-host/document-service',
      timeout: 'config/application/timeout',
      version: 'config/application/version',
    }
  },
  logs: {
    defaultAjaxUrl: "/" + context + "/client.logger"
  },
  serverApiEndpoints: {
    // plop import the server config here

  },
  socketIO: {
    path: context + '/io/socket'
  },
  amqp: {
    queueName: 'frontend.queue',
    exchangeName: 'frontend.exchange',
    options: {
      host: '192.168.1.163',
      port: 5672,
      login: 'guest',
      password: 'guest',
      connectionTimeout: 10000,
      vhost: '/',
      noDelay: true,
      ssl: {
        enabled: false
      }
    }
  }
});
