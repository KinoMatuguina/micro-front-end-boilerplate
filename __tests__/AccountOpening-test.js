
import rp from 'request-promise';
import ServerConfig from '../app/config/ServerConfig';

const tips = ServerConfig.serverApiEndpoints.tips;
const systemStatus = ServerConfig.serverApiEndpoints.systemStatus;
const nationality = ServerConfig.serverApiEndpoints.nationality;
const country = ServerConfig.serverApiEndpoints.country;
const province = ServerConfig.serverApiEndpoints.province;
const city = ServerConfig.serverApiEndpoints.city;
const branch = ServerConfig.serverApiEndpoints.branch;
const connectorCd = ServerConfig.serverApiEndpoints.connectorCd;
const gross = ServerConfig.serverApiEndpoints.gross;
const acceptableId = ServerConfig.serverApiEndpoints.acceptableId;
const natureOfBusiness = ServerConfig.serverApiEndpoints.natureOfBusiness;
const natureOfEmployment = ServerConfig.serverApiEndpoints.natureOfEmployment;
const sourceOfFunds = ServerConfig.serverApiEndpoints.sourceOfFunds;



describe('Maintinance services', () => {

 it('should load tips and match on snapshots', () => {
 	var tree = {}
 	return rp({
		  method: "GET",
		  uri: tips,
		  body: {},
		  resolveWithFullResponse: true,
		  json: true})
	    .then(function (response) {
	    	expect.assertions(1);
	    	tree = response.body
	    	expect(tree).toMatchSnapshot();
	    })
	 })

 it('should load systemStatus and match on snapshots', () => {
 	var tree = {}
 	return rp({
		  method: "GET",
		  uri: systemStatus,
		  body: {},
		  resolveWithFullResponse: true,
		  json: true})
	    .then(function (response) {
	    	expect.assertions(1);
	    	tree = response.body
	    	expect(tree).toMatchSnapshot();
	    })
 })

 it('should load nationality and match on snapshots', () => {
 	var tree = {}
 	return rp({
		  method: "GET",
		  uri: nationality,
		  body: {},
		  resolveWithFullResponse: true,
		  json: true})
	    .then(function (response) {
	    	expect.assertions(1);
	    	tree = response.body
	    	expect(tree).toMatchSnapshot();
	    })
 })

 it('should load country and match on snapshots', () => {
 	var tree = {}
 	return rp({
		  method: "GET",
		  uri: country,
		  body: {},
		  resolveWithFullResponse: true,
		  json: true})
	    .then(function (response) {
	    	expect.assertions(1);
	    	tree = response.body
	    	expect(tree).toMatchSnapshot();
	    })
 })

 it('should load natureOfBusiness and match on snapshots', () => {
 	var tree = {}
 	return rp({
		  method: "GET",
		  uri: natureOfBusiness,
		  body: {},
		  resolveWithFullResponse: true,
		  json: true})
	    .then(function (response) {
	    	expect.assertions(1);
	    	tree = response.body
	    	expect(tree).toMatchSnapshot();
	    })
 })

 it('should load natureOfEmployment and match on snapshots', () => {
 	var tree = {}
 	return rp({
		  method: "GET",
		  uri: natureOfEmployment,
		  body: {},
		  resolveWithFullResponse: true,
		  json: true})
	    .then(function (response) {
	    	expect.assertions(1);
	    	tree = response.body
	    	expect(tree).toMatchSnapshot();
	    })
 })

 it('should load sourceOfFunds and match on snapshots', () => {
 	var tree = {}
 	return rp({
		  method: "GET",
		  uri: sourceOfFunds,
		  body: {},
		  resolveWithFullResponse: true,
		  json: true})
	    .then(function (response) {
	    	expect.assertions(1);
	    	tree = response.body
	    	expect(tree).toMatchSnapshot();
	    })
 })

 it('should load task and match on snapshots', () => {
 	var tree = {}
 	return rp({
	  method: "GET",
	  uri: "http://192.168.1.249:8088/task/findTaskByBranchCodeUserRole?page=0&count=10&order=ASC&sort=lastUpdatedDate&branchCode=0001&financialInstitution=BPI%20FAMILY%20SAVINGS&userRole=MAKER2&isViewAll=true",
	  body: {},
	  resolveWithFullResponse: true,
	  json: true})
    .then(function (response) {
    	expect.assertions(1);
    	tree = response.body
    	expect(tree).toMatchSnapshot();
    })
 })

 it('should load financial institution image and match on snapshots', () => {
 	var tree = {}
 	return rp({
		  method: "GET",
		  uri: "http://192.168.1.249:8053/logo/findLogoByFinancialInstitution?financialInstitution=BPI%20FAMILY%20SAVINGS",
		  body: {},
		  resolveWithFullResponse: true,
		  json: true})
	    .then(function (response) {
	    	expect.assertions(1);
	    	tree = response.body
	    	expect(tree).toMatchSnapshot();
	    })
 })

it('should load bundle and match on snapshots', () => {
 	var tree = {}
 	return rp({
		  method: "GET",
		  uri: "http://192.168.1.249:8088/bundle/findAllBundleByFinancialInstitution?financialInstitution=BPI%20FAMILY%20SAVINGS",
		  body: {},
		  resolveWithFullResponse: true,
		  json: true})
	    .then(function (response) {
	    	expect.assertions(1);
	    	tree = response.body
	    	expect(tree).toMatchSnapshot();
	    })
 })

it('should load product and match on snapshots', () => {
 	var tree = {}
 	return rp({
		  method: "GET",
		  uri: "http://192.168.1.249:8088/product/findAllProductByBundleIdFinancialInstitution?bundleId=001&financialInstitution=BPI%20FAMILY%20SAVINGS",
		  body: {},
		  resolveWithFullResponse: true,
		  json: true})
	    .then(function (response) {
	    	expect.assertions(1);
	    	tree = response.body
	    	expect(tree).toMatchSnapshot();
	    })
 })

})
