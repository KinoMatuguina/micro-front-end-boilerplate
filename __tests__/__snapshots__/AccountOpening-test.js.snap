// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Maintinance services should load bundle and match on snapshots 1`] = `
Array [
  Object {
    "bundleName": "BUNDLE 1",
    "bundlePhoto": null,
    "financialInstitution": "BPI FAMILY SAVINGS",
    "id": "001",
    "isOfferred": true,
  },
  Object {
    "bundleName": "BUNDLE 2",
    "bundlePhoto": null,
    "financialInstitution": "BPI FAMILY SAVINGS",
    "id": "002",
    "isOfferred": true,
  },
  Object {
    "bundleName": "BUNDLE 3",
    "bundlePhoto": null,
    "financialInstitution": "BPI FAMILY SAVINGS",
    "id": "003",
    "isOfferred": true,
  },
]
`;

exports[`Maintinance services should load country and match on snapshots 1`] = `
Array [
  Object {
    "auditUserCd": null,
    "cd": "PHILIPPINES",
    "correlationId": null,
    "countryCd": "63",
    "createdBy": "SYSTEM",
    "createdDate": 1494829679231,
    "del": false,
    "dscp": "PHILIPPINES",
    "id": "001",
    "updatedBy": null,
    "updatedDate": null,
    "version": 0,
  },
]
`;

exports[`Maintinance services should load financial institution image and match on snapshots 1`] = `
Array [
  Object {
    "financialInstitution": "BPI FAMILY SAVINGS",
    "logo": null,
  },
]
`;

exports[`Maintinance services should load nationality and match on snapshots 1`] = `
Array [
  Object {
    "auditUserCd": null,
    "cd": "FIL",
    "correlationId": null,
    "createdBy": "SYSTEM",
    "createdDate": 1494843868152,
    "del": false,
    "dscp": "FILIPINO",
    "id": "001",
    "updatedBy": null,
    "updatedDate": null,
    "version": 0,
  },
  Object {
    "auditUserCd": null,
    "cd": "AME",
    "correlationId": null,
    "createdBy": "SYSTEM",
    "createdDate": 1494843868165,
    "del": false,
    "dscp": "AMERICAN",
    "id": "002",
    "updatedBy": null,
    "updatedDate": null,
    "version": 0,
  },
]
`;

exports[`Maintinance services should load natureOfBusiness and match on snapshots 1`] = `
Array [
  Object {
    "auditUserCd": null,
    "cd": "ITSS",
    "correlationId": null,
    "createdBy": "SYSTEM",
    "createdDate": 1494998556308,
    "del": false,
    "dscp": "IT Services and Supply ",
    "id": "001",
    "updatedBy": null,
    "updatedDate": null,
    "version": 0,
  },
  Object {
    "auditUserCd": null,
    "cd": "HRSRQ",
    "correlationId": null,
    "createdBy": "SYSTEM",
    "createdDate": 1494998556315,
    "del": false,
    "dscp": "Human Resource Services, Recruitment and Qualifications",
    "id": "002",
    "updatedBy": null,
    "updatedDate": null,
    "version": 0,
  },
  Object {
    "auditUserCd": null,
    "cd": "MP",
    "correlationId": null,
    "createdBy": "SYSTEM",
    "createdDate": 1494998556326,
    "del": false,
    "dscp": "Marketing and Promotion ",
    "id": "003",
    "updatedBy": null,
    "updatedDate": null,
    "version": 0,
  },
  Object {
    "auditUserCd": null,
    "cd": "TELECOM",
    "correlationId": null,
    "createdBy": "SYSTEM",
    "createdDate": 1494998556338,
    "del": false,
    "dscp": "Telecommunications ",
    "id": "004",
    "updatedBy": null,
    "updatedDate": null,
    "version": 0,
  },
  Object {
    "auditUserCd": null,
    "cd": "PS",
    "correlationId": null,
    "createdBy": "SYSTEM",
    "createdDate": 1494998556344,
    "del": false,
    "dscp": "Professional Services",
    "id": "005",
    "updatedBy": null,
    "updatedDate": null,
    "version": 0,
  },
]
`;

exports[`Maintinance services should load natureOfEmployment and match on snapshots 1`] = `
Array [
  Object {
    "auditUserCd": null,
    "cd": "S",
    "correlationId": null,
    "createdBy": "SYSTEM",
    "createdDate": 1495085761751,
    "del": false,
    "dscp": "Self-Employed",
    "id": "001",
    "income": null,
    "updatedBy": null,
    "updatedDate": null,
    "version": 0,
  },
  Object {
    "auditUserCd": null,
    "cd": "C",
    "correlationId": null,
    "createdBy": "SYSTEM",
    "createdDate": 1495085761828,
    "del": false,
    "dscp": "Casual",
    "id": "002",
    "income": null,
    "updatedBy": null,
    "updatedDate": null,
    "version": 0,
  },
  Object {
    "auditUserCd": null,
    "cd": "R",
    "correlationId": null,
    "createdBy": "SYSTEM",
    "createdDate": 1495085761838,
    "del": false,
    "dscp": "Regular",
    "id": "003",
    "income": null,
    "updatedBy": null,
    "updatedDate": null,
    "version": 0,
  },
]
`;

exports[`Maintinance services should load product and match on snapshots 1`] = `
Array [
  Object {
    "accountType": "Single",
    "classification": "Savings",
    "currencyCode": "001",
    "features": null,
    "id": "001",
    "media": "Debit Card",
    "productGroup": "Deposit",
    "productImage": null,
    "productName": "KIDDY SAVER W\\\\/ ATM",
  },
  Object {
    "accountType": "Single",
    "classification": "Savings",
    "currencyCode": "001",
    "features": null,
    "id": "002",
    "media": "Debit Card",
    "productGroup": "Deposit",
    "productImage": null,
    "productName": "PLATINUM SAVINGS W\\\\/ ATM",
  },
]
`;

exports[`Maintinance services should load sourceOfFunds and match on snapshots 1`] = `
Array [
  Object {
    "auditUserCd": null,
    "cd": "PAYROLL",
    "correlationId": null,
    "createdBy": "SYSTEM",
    "createdDate": 1495076458771,
    "del": false,
    "dscp": "Payroll",
    "id": "001",
    "updatedBy": null,
    "updatedDate": null,
    "version": 0,
  },
  Object {
    "auditUserCd": null,
    "cd": "BUSINESS",
    "correlationId": null,
    "createdBy": "SYSTEM",
    "createdDate": 1495076458779,
    "del": false,
    "dscp": "Business",
    "id": "002",
    "updatedBy": null,
    "updatedDate": null,
    "version": 0,
  },
  Object {
    "auditUserCd": null,
    "cd": "STOCKS",
    "correlationId": null,
    "createdBy": "SYSTEM",
    "createdDate": 1495076458789,
    "del": false,
    "dscp": "Dividend Stocks",
    "id": "003",
    "updatedBy": null,
    "updatedDate": null,
    "version": 0,
  },
]
`;

exports[`Maintinance services should load systemStatus and match on snapshots 1`] = `
Array [
  Object {
    "advisoryDt": 1494835456844,
    "auditUserCd": null,
    "correlationId": null,
    "createdBy": "SYSTEM",
    "createdDate": 1494835456844,
    "del": false,
    "id": "002",
    "msg": "MESSAGE ONE",
    "updatedBy": null,
    "updatedDate": null,
    "version": 0,
  },
  Object {
    "advisoryDt": 1494835475897,
    "auditUserCd": null,
    "correlationId": null,
    "createdBy": "SYSTEM",
    "createdDate": 1494835475897,
    "del": false,
    "id": "001",
    "msg": "MESSAGE TWO",
    "updatedBy": null,
    "updatedDate": null,
    "version": 0,
  },
]
`;

exports[`Maintinance services should load task and match on snapshots 1`] = `Array []`;

exports[`Maintinance services should load tips and match on snapshots 1`] = `
Array [
  Object {
    "auditUserCd": null,
    "correlationId": null,
    "createdBy": "SYSTEM",
    "createdDate": 1494835158717,
    "del": false,
    "id": "001",
    "pageName": "PAGE ONE",
    "tips": "TIPS 1",
    "updatedBy": null,
    "updatedDate": null,
    "version": 0,
  },
  Object {
    "auditUserCd": null,
    "correlationId": null,
    "createdBy": "SYSTEM",
    "createdDate": 1494835238102,
    "del": false,
    "id": "002",
    "pageName": "PAGE TWO",
    "tips": "TIPS 2",
    "updatedBy": null,
    "updatedDate": null,
    "version": 0,
  },
]
`;
